x-pack/plugins/apm/server/feature.ts

```diff
@@ -58,5 +63,43 @@ export const APM_FEATURE = {
   },
 };
 
-export const APM_SERVICE_MAPS_FEATURE_NAME = 'APM service maps';
-export const APM_SERVICE_MAPS_LICENSE_TYPE = 'platinum';
+interface Feature {
+  name: string;
+  license: LicenseType;
+}
+type FeatureName = 'serviceMaps' | 'ml' | 'customLinks';
+export const features: Record<FeatureName, Feature> = {
+  serviceMaps: {
+    name: 'APM service maps',
+    license: 'platinum',
+  },
+  ml: {
+    name: 'APM machine learning',
+    license: 'platinum',
+  },
+  customLinks: {
+    name: 'APM custom links',
+    license: 'gold',
+  },
+};
+
+export function registerFeaturesUsage({
+  licensingPlugin,
+}: {
+  licensingPlugin: LicensingPluginSetup;
+}) {
+  Object.values(features).forEach(({ name, license }) => {
+    licensingPlugin.featureUsage.register(name, license);
+  });
+}
+
+export function notifyFeatureUsage({
+  licensingPlugin,
+  featureName,
+}: {
+  licensingPlugin: LicensingRequestHandlerContext;
+  featureName: FeatureName;
+}) {
+  const feature = features[featureName];
+  licensingPlugin.featureUsage.notifyUsage(feature.name);
+}
```

Comment:
@sqren what do you think of this? too much?

Reply:
Looks good 👍